Use selected text for gist, if available

gadenbuie opened the pull request:
Hey @MilesMcBain, thanks for putting this package together!

I made a small change to `gistfo()` to use the currently selected text for the gist (if available), falling back to the source file if no text is selected. While I was poking around I also fixed a typo and updated the addins dropdown list menu items.

No pressure to merge or keep all of my changes, just thought I'd share my tweaks.

gadenbuie committed:
If text is selected, limit gist/carbon to selection

gadenbuie committed:
spelling: unititled -> untitled

gadenbuie committed:
Change description and name of RStudio addins

MilesMcBain commented:
I like this change, thanks!

A couple of things to consider before merge:
* Should the name of the gist be different for selections? It currently reflects the file. I'm thinking maybe it should tack "_selection" on the end.
* The fallback behavior probably needs to be explained a bit more explicitly in the README.

gadenbuie committed:
Add "selection" to gist name and update docs

- If gist from selection, add "selection" between project and file names
- Updated function documentation
- Added titles to separate function title & description
- Updated package description
- Updated README
- Set Roxygen to use markdown in documentation
- Regenerated docs with devtools::document()

gadenbuie commented:
Thanks! So I updated the documentation, mostly minor changes to add in the "selected text" bit. Those changes updated the README, function docs and the package description. Let me know though if any of it's unclear.

Rather than adding "_selection" at the end, I inserted it between "project" and "filename", e.g. `RStudio_gistfo_selection_gistfo.R`. Mostly because this was easier and doesn't involve  parsing the filename to insert `_selection` before the file extension. (Not that I'm afraid of regular expressions :stuck_out_tongue_winking_eye: it just seemed like overkill for this situation.)

MilesMcBain commented:
Oooooh look at that commit msg. A+ pull. Thanks!
